projects/plugins/debug-helper/modules/class-idc-simulator.php

```diff
@@ -0,0 +1,238 @@
+			<tbody>
+			<tr>
+				<th scope="row">IDC Simulation</th>
+				<td>
+					<fieldset><legend class="screen-reader-text"><span>Site URL Spoofing</span></legend>
+					<label><input type="radio" name="idc_simulation" value="1" <?php echo ( $settings['idc_simulation'] ? 'checked="checked"' : '' ); ?>> enabled</label><br>
+					<label><input type="radio" name="idc_simulation" value="0" <?php echo ( ! $settings['idc_simulation'] ? 'checked="checked"' : '' ); ?>> disabled</label><br>
+					</fieldset>
+				</td>
+			</tr>
+			<tr>
+				<th scope="row"><label for="idc_siteurl">Spoof URL</label></th>
+				<td><input name="idc_siteurl" type="text" id="blogname" value="<?php echo esc_attr( $settings['idc_siteurl'] ); ?>" class="regular-text"></td>
+			</tr>
+
+			</tbody>
+			</table>
+
+			<input type="hidden" name="action" value="store_current_options">
+			<?php wp_nonce_field( 'store-current-options' ); ?>
+			<input type="submit" value="Store these options" class="button button-primary">
+		</form>
+
+		<hr>
+		<?php
+	}
+
+	/**
+	 * Store options.
+	 */
+	public function admin_post_store_current_options() {
+		check_admin_referer( 'store-current-options' );
+		$this->notice_type = 'store-options';
+
+		update_option(
+			self::STORED_OPTIONS_KEY,
+			array(
+				'idc_siteurl'    => $_POST['idc_siteurl'],
+				'idc_simulation' => $_POST['idc_simulation'],
+			)
+		);
+
+		$this->admin_post_redirect_referrer();
+	}
+
+	/**
+	 * Retrieves the stored IDC options.
+	 *
+	 * @return array
+	 */
+	public function get_stored_settings() {
+		return get_option( self::STORED_OPTIONS_KEY );
+	}
+
+	/**
+	 * Clears all stored option values.
+	 */
+	public function clear_stored_options() {
+		delete_option( self::STORED_OPTIONS_KEY );
+	}
+
+	/**
+	 * Just redirects back to the referrer. Keeping it DRY.
+	 */
+	public function admin_post_redirect_referrer() {
+		if ( wp_get_referer() ) {
+			wp_safe_redirect(
+				add_query_arg(
+					array(
+						'notice' => $this->notice_type,
+						'nonce'  => wp_create_nonce( 'jetpack_debug_broken_token_admin_notice' ),
+					),
+					wp_get_referer()
+				)
+			);
+		} else {
+			wp_safe_redirect( get_home_url() );
+		}
+	}
+
+	/**
+	 * Displays an admin notice...
+	 */
+	public function render_admin_notice() {
+		switch ( $this->notice_type ) {
+			case 'store-options':
+				$message = 'IDC Simulation settings saved successfully!';
+				break;
+			default:
+				$message = 'Setting saved!';
+				break;
+		}
+
+		printf( '<div class="notice notice-success"><p>%s</p></div>', esc_html( $message ) );
+	}
+}
+
+add_action( 'plugins_loaded', 'register_idc_simulator', 1000 );
+
+/**
+ * Load the simulator
+ */
+function register_idc_simulator() {
+	if ( class_exists( 'Jetpack_Options' ) ) {
+		new IDC_Simulator();
+	} else {
+		add_action( 'admin_notices', 'idc_simulator_jetpack_not_active' );
+	}
+}
+
+/**
+ * Notice for if Jetpack is not active.
+ */
+function idc_siulator_jetpack_not_active() {
```

Comment:
Just a typo:

```suggestion
function idc_simulator_jetpack_not_active() {
```

Reply:
Ouch, I didn't see you submitted a suggestion - I have fixed it in another commit, but thanks anyway!